internal/capabilities/detector.go

```diff
@@ -38,10 +38,12 @@ func DetectCapabilities(jc config.JobConfiguration, jobServer JobServerInterface
 	accounts := jc.GetStringSlice("twitter_accounts", nil)
 	apiKeys := jc.GetStringSlice("twitter_api_keys", nil)
 	apifyApiKey := jc.GetString("apify_api_key", "")
+	geminiApiKey := jc.GetString("gemini_api_key", "")
```

Comment:
Can we do the cast here instead of below, so that `geminiApiKey` is always an `LlmApiKey`?

One of the principles behind using these types is creating them as early as possible, so that the compiler will tell you when you're mixing things up and you can call their methods everywhere without having to cast again. So the moment you get it you cast it. To be even more strict, you could cast it when adding it to the config (that's what you do if you use a structured config).